### PR TITLE
DO NOT MERGE Spike/lazy module loading

### PR DESCRIPTION
**Pull request by rfreitas**
# Problem
Reduce bloat by splitting not always needed modules

# Exploration

1. Updated Webpack to version 3
2. Followed the guide here for code splitting https://webpack.js.org/guides/code-splitting/
3. Imported `react-phone-number-input` lazyly on the first step

### CODE REVIEW

**Comment by rfreitas** [comment]
Travis automatic deployment: https://staging-220-pr-onfido-sdk-ui-onfido.surge.sh

File: webpack.config.babel.js [hunk 1]
```diff
@@ -136,17 +136,27 @@ const configDist = {
     libraryTarget: 'umd',
     path: `${__dirname}/dist`,
     publicPath: '/',
-    filename: 'onfido.min.js'
+    filename: 'onfido.min.js',
+    chunkFilename: '[name].bundle.js',
```
Comment (hjheath) [inline webpack.config.babel.js]: what is `name` here?

File: src/components/PhoneNumberInput/lazy.js [hunk 2]
```diff
@@ -0,0 +1,18 @@
+import { h, Component } from 'preact'
+
+let Empty = () => <div>Not here</div>
+
+class PhoneNumberInputLazy extends Component {
+  constructor(props){
+    super(props)
+    this.state = {component: Empty }
+
+    import(/* webpackChunkName: "cross-device" */ './index.js').then(component => {
```
Comment (hjheath) [inline src/components/PhoneNumberInput/lazy.js]: so this triggers a download of another bundle?

How does the browser know what to download?

File: src/components/PhoneNumberInput/index.js [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+import 'react-phone-number-input/rrui.css'
+import 'react-phone-number-input/style.css'
+
+class PhoneNumberWithState extends Component {
+  onChange = (value) => this.setState({value})
+  render = ()=>
+    <PhoneNumber onChange={this.onChange} value={this.state.value}/>
+}
+
+export default PhoneNumberWithState
```
Comment (stefaniacardenas) [inline src/components/PhoneNumberInput/index.js]: Where are you using this?

File: src/components/PhoneNumberInput/lazy.js (hunk 2 above)
Comment (rfreitas) [reply src/components/PhoneNumberInput/lazy.js]: this is not lazy loading, only code splitting. It's up to to the developer to still include the secondary js bundle in the html. The advantage is that they can load after the page load (by using `async`). Lazy loading requires to host the js bundles.

File: webpack.config.babel.js (hunk 1 above)
Comment (rfreitas) [reply webpack.config.babel.js]: what is defined in the comment of `webpackChunkName`

File: src/components/PhoneNumberInput/index.js (hunk 3 above)
Comment (stefaniacardenas) [reply src/components/PhoneNumberInput/index.js]: Don't worry, I can see it now 👍

**Pull request closed without merging**